### PR TITLE
HOSTEDCP-1798: Refactor multi-arch validation algorithm

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Refactors the algorithm performing different multi-arch validations similar to [this](https://excalidraw.com/#json=QeumxPAwNmXPFxe4GjuWt,I_h08v5-INft4lOLYO5Mcg).

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1778](https://issues.redhat.com/browse/HOSTEDCP-1778)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by bryan-cox** [comment]
I did not implement this block from the drawing as I don't recall why we originally needed it
```
 if multi-arch flag set 
        warn user multi-arch image not used,
        block hc creation
```

File: cmd/cluster/aws/create.go [hunk 1]
```diff
@@ -504,3 +499,46 @@ func validateAWSOptions(ctx context.Context, opts *core.CreateOptions, awsOpts *
...
+		}
+
+		validMultiArchImage, err = registryclient.IsMultiArchManifestList(ctx, opts.ReleaseImage, pullSecret)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check if a release stream was provided instead and its multi-arch
+	if len(opts.ReleaseStream) > 0 && !strings.Contains(opts.ReleaseStream, "multi") {
```
Comment (muraee) [inline cmd/cluster/aws/create.go:520]:
```suggestion
	if len(opts.ReleaseStream) > 0 && strings.Contains(opts.ReleaseStream, "multi") {
```
shouldn't this be a positive check or did I misunderstand it?

File: cmd/cluster/aws/create.go [hunk 2]
```diff
@@ -504,3 +499,46 @@ func validateAWSOptions(ctx context.Context, opts *core.CreateOptions, awsOpts *
 
 	return nil
 }
+
+func validateMultiArch(ctx context.Context, o *RawCreateOptions, opts *core.CreateOptions) error {
+	var validMultiArchImage bool
```
Comment (muraee) [inline cmd/cluster/aws/create.go:504]:
```suggestion
	validMultiArchImage := false
```
nit: make it more explicit

File: cmd/cluster/aws/create.go (hunk 1 above)
Comment (bryan-cox) [reply cmd/cluster/aws/create.go:520]: Yes 🤦🏻

**Pull request merged**